src/lib/shadow-shim-helper-rs/src/shim_event.rs

```diff
@@ -27,13 +27,14 @@ pub struct ShimEventAddThreadReq {
     pub ipc_block: ShMemBlockSerialized,
 }
 
-/// Data for [`ShimEventToShim::Start`]
-/// TODO: Consider splitting into multiple messages
+/// Data for [`ShimEventToShadow::StartReq`]
 #[derive(Copy, Clone, Debug, VirtualAddressSpaceIndependent)]
 #[repr(C)]
-pub struct ShimEventStart {
-    pub thread_shmem_block: ShMemBlockSerialized,
-    pub process_shmem_block: ShMemBlockSerialized,
+pub struct ShimEventStartReq {
+    /// Pointer to be initd by Shadow. Required.
+    pub thread_shmem_block_to_init: ForeignPtr<ShMemBlockSerialized>,
+    /// Pointer to be initd by Shadow. Optional.
+    pub process_shmem_block_to_init: ForeignPtr<ShMemBlockSerialized>,
```

Comment:
Maybe clarify that these are pointers to plugin memory. Since these pointers are shared with both shadow and the shim, it's not immediately clear which is the "foreign" memory space (it's not really a "foreign pointer" when you're accessing these fields in the shim).